gprofiler/main.py

```diff
@@ -313,26 +313,17 @@ def _snapshot(self) -> None:
             self._generate_output_files(merged_result, local_start_time, local_end_time)
 
         if self._client:
-            try:
-                response_dict = self._client.submit_profile(
-                    local_start_time,
-                    local_end_time,
-                    merged_result,
-                    total_samples,
-                    self._profile_api_version,
-                    self._spawn_time,
-                    metrics,
-                    self._gpid,
-                )
-                self._gpid = response_dict.get("gpid", "")
```

Comment:
`self._gpid` is saved across sessions - you can return it from `_submit_profile_logged` and write it back to `self._gpid` here.